### ISSUE TITLE
Block HTTP messages using LF only as line terminators instead of full CRLF

### ISSUE THREAD

**Issue by krizhanovsky** [natsys/tempesta#444]
RFC 7230 allows, but doesn't require to support, messages with lines terminated with LF only, without CR. While Tempesta FW processes lines termination with LF only as well as with full CRLF, it requires more complex (and slower) parsing and complicates HTTP messages transformation. Currently, there are bugs in cutoff logic: https://github.com/natsys/tempesta/blob/master/tempesta_fw/http_msg.c#L447 and https://github.com/natsys/tempesta/blob/master/tempesta_fw/http_msg.c#L486 - header trailers are considered 2 bytes in size.

So we should support CRLF only terminations.

**Comment by krizhanovsky** [comment]
Commit https://github.com/natsys/tempesta/pull/340/commits/3ffc05323776f92d16dfa6b87c81c403f485cb15 relates to the issue.

**Comment by milabs** [comment]
RFC 2616 says that:

~~~
19.3 Tolerant Applications

   Clients SHOULD be tolerant in parsing the Status-Line and servers
   tolerant when parsing the Request-Line. In particular, they SHOULD
   accept any amount of SP or HT characters between fields, even though
   only a single SP is required.

   The line terminator for message-header fields is the sequence CRLF.
   However, we recommend that applications, when parsing such headers,
   recognize a single LF as a line terminator and ignore the leading CR.
~~~

As for now, RFC 2616 obsoleted by RFC 7230. But there still many implementations which doesn't follow the strict CRLF rules. So, is it useful for the Tempesta not to support them?

**Comment by krizhanovsky** [comment]
Frankly speaking I didn't see any real client using LF only as line terminator.

**Comment by milabs** [comment]
There is a question about validness of the `\r...\r\r\r\n` sequence. Google and Yandex has no united meaning of how to deal with it:

1. Google
~~~
$ echo -en "GET / HTTP/1.1\r\nHost: www.google.com\r\n\r\n" | socat - TCP:www.google.com:80
HTTP/1.1 302 Found
...

$ echo -en "GET / HTTP/1.1\r\r\r\r\r\nHost: www.google.com\r\n\r\n" | socat - TCP:www.google.com:80
HTTP/1.1 302 Found
...
~~~

2. Yandex
~~~
$ echo -en "GET / HTTP/1.1\r\nHost: www.yandex.ru\r\n\r\n" | socat - TCP:www.yandex.ru:80
HTTP/1.1 302 Found

$ echo -en "GET / HTTP/1.1\r\r\r\r\r\nHost: www.yandex.ru\r\n\r\n" | socat - TCP:www.yandex.ru:80
HTTP/1.1 400 Bad request
~~~

So, the question is how Tempesta should handle such issues?

**Comment by krizhanovsky** [comment]
It seems RFC doesn't allow multiple CRs. Also I didn't see real application sending many CRs, usually this is hand crafted things. Meantime, handling them makes original bugs from https://github.com/natsys/tempesta/issues/444#issue-144323096 harder to fix. Thus, I think we should prohibit such requests.

**Comment by krizhanovsky** [comment]
Please also fix https://github.com/natsys/tempesta/blob/master/tempesta_fw/http_msg.c#L484 : the length of ```orig_hdr``` without CRLF is compared with ```hdr``` with CRLF, so there are cases when we go by slow path in the headers substitution.

**Comment by milabs** [comment]
Fixed by 04ca164fc5fbb55b468f1e425b5e47dacbfcff08
